mlflow/pyfunc/__init__.py

```diff
@@ -2097,6 +2097,20 @@ def spark_udf(
     # can't access NFS, so it needs to use `spark.addArtifact`.
     use_dbconnect_artifact = is_dbconnect_mode and not is_in_databricks_shared_cluster_runtime()
 
+    if use_dbconnect_artifact:
+        udf_sandbox_info = get_dbconnect_udf_sandbox_info()
+        if Version(udf_sandbox_info.mlflow_version) < Version("2.18.0"):
+            raise MlflowException(
+                "Using 'mlflow.pyfunc.spark_udf' in Databricks Serverless or in remote "
+                "Databricks Connect requires UDF sandbox image installed with MLflow "
+                "of version >= 2.18."
+            )
```

Comment:
The UDF execution needs `DBConnectArtifactCache` related code which only exists in  MLflow >= 2.18